gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -420,5 +420,188 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
+  ZoneInfo getZone(BigInteger zoneId, ZoneOption... options);
+
+  /**
+   * Lists the zones inside the project.
+   *
+   * <p>This method returns zones in an unspecified order. New zones do not necessarily appear at
+   * the end of the list. Use {@link ZoneListOption} to restrict the listing to a domain name, set
+   * page size, and set page token.
+   *
+   * @return a page of zones
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/list">Cloud DNS Managed Zones:
+   * list</a>
+   */
+  Page<Zone> listZones(ZoneListOption... options);
+
+  /**
+   * Deletes an existing zone identified by name. Returns {@code true} if the zone was successfully
+   * deleted and {@code false} otherwise.
+   *
+   * @return {@code true} if zone was found and deleted and {@code false} otherwise
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/delete">Cloud DNS Managed Zones:
+   * delete</a>
+   */
+  boolean delete(String zoneName); // delete does not admit any options
+
+  /**
+   * Deletes an existing zone identified by id. Returns {@code true} if the zone was successfully
+   * deleted and {@code false} otherwise.
+   *
+   * @return {@code true} if zone was found and deleted and {@code false} otherwise
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/delete">Cloud DNS Managed Zones:
+   * delete</a>
+   */
+  boolean delete(BigInteger zoneId); // delete does not admit any options
+
+  /**
+   * Lists the DNS records in the zone identified by name.
+   *
+   * <p>The fields to be returned, page size and page tokens can be specified using {@link
+   * DnsRecordListOption}s.
+   *
+   * @throws DnsException upon failure or if the zone cannot be found
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(String zoneName, DnsRecordListOption... options);
+
+  /**
+   * Lists the DNS records in the zone identified by ID.
+   *
+   * <p>The fields to be returned, page size and page tokens can be specified using {@link
+   * DnsRecordListOption}s.
+   *
+   * @throws DnsException upon failure or if the zone cannot be found
+   * @see <a href="https://cloud.google.com/dns/api/v1/resourceRecordSets/list">Cloud DNS
+   * ResourceRecordSets: list</a>
+   */
+  Page<DnsRecord> listDnsRecords(BigInteger zoneId, DnsRecordListOption... options);
+
+  /**
+   * Retrieves the information about the current project. The returned fields can be optionally
+   * restricted by specifying {@link ProjectOption}s.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/projects/get">Cloud DNS Projects: get</a>
+   */
+  ProjectInfo getProjectInfo(ProjectOption... fields);
+
+  /**
+   * Returns the current project id.
+   */
+  String getProjectId();
```

Comment:
I don't think we need `getProjectNumber` and `getProjectId` in this level.

However, at some point we should consider having them as part of the `Service` api (and probably implemented by the `BaseService`).